backend/data/users.js

```diff
@@ -61,6 +63,205 @@ async function checkUser(email, password) {
+  if (updatedInfo.modifiedCount === 0) throw 'Failed to add course to teacher!';
+
+
+  //upon successful add, return the name of the teacher for output
+  return tempTeacher.firstName + " " + tempTeacher.lastName;
+}
+
+//Note: Takes in courseId and userId as strings. 
+async function drop(courseId, userId) {
+  //error check inputs
+  error.str(courseId);
+  error.str(userId);
+
+  //retrieve the users collection
+  const userCollection = await users();
+
+  //check if the provided userId is a valid ObjectId
+  if(!ObjectId.isValid(userId)) throw 'Provided ID not proper format for ObjectID!';
+  //convert the ID string to an ObjectID
+  let objId = ObjectId(userId);
+
+  //retrieve the original user information
+  let tempUser = await userCollection.findOne({ _id: objId });
+
+  //the above call will result in null if the given ID doesn't exist in the database
+  if(tempUser === null) throw 'User with that ID is not in database!';
+
+  //Make sure there's at least 1 course in the user's "classes" array
+  if(tempUser.classes.length == 0) throw "'Cannot drop a course from a user that doesn't have any courses!";
+
+  //boolean to track if the course was found in the user's "classes" array
+  let found = false;
+  
+  //find the course in the user's "classes" array
+  for(let course of tempUser.classes)  {
+    //once the course has been found
+    if(course._id == courseId)  {
+      found = true;
+      //edge case: user is dropping the only class they're enrolled in 
+      if(tempUser.classes.length == 1)
+        tempUser.classes.pop();
+      //otherwise, remove the course from the array
+      else {
+        let ind = tempUser.classes.indexOf(course);
+        tempUser.classes.splice(ind, 1);
+      }
+    }
+  }
+
+  //Throw an error when trying to drop a course that the user wasn't involved in
+  if(found == false)  throw 'User was not enrolled-in or teaching that course!';
+
+  //"push" the updated course info to the same ID in the database
+  const updatedInfo = await userCollection.updateOne(
+      {_id: objId},
+      { $set: tempUser}
+  );
+
+  //check that the update succeeded
+  if (updatedInfo.modifiedCount === 0) throw 'Failed to drop course!';
+
+
+  //upon successful add, return the name of the student for output
+  return tempUser.firstName + " " + tempUser.lastName;
+}
+
+async function getCourses(userId)  {
+  //error check input
+  error.str(userId);
+
+  //retrieve the user collection
+  const userCollection = await users();
+
+  //check if the provided userId is a valid ObjectId
+  if(!ObjectId.isValid(userId)) throw 'Provided ID not proper format for ObjectID!';
+  //convert the ID string to an ObjectID
+  let objId = ObjectId(userId);
+
+  //retrieve the user's information
+  let tempUser = await userCollection.findOne({ _id: objId });
+
+  //the above call will result in null if the given ID doesn't exist in the database
+  if(tempUser === null) throw 'User with that ID is not in database!';
```

Comment:
You can check against null and undefined by using a bang `!`. You can use this for all your occurrences of checking `=== null`.

```suggestion
  if(!tempUser) throw 'User with that ID is not in database!';
```